src/Mirza/SupplyChain/PopulateUtils.hs

```diff
@@ -120,24 +121,30 @@ insertAndAuth scsUrl brUrl auth locMap ht (entity:entities) = do
       (Entity name companyPrefix bizName locations (KeyPairPaths _ pubKeyPath)) = entity
       [newUserSCS] = genMultipleUsersSCS "citrusTest" 1 [name] [companyPrefix]
       newBiz = BT.NewBusiness companyPrefix bizName
-  [newUserBR] <- GenBR.generateMultipleUsers "citrusTest" 1 [companyPrefix]
+  [newUserBR] <- GenBR.generateMultipleUsers "citrusTest" 1 [name] [companyPrefix]
+  let userAuth = BasicAuthData
+                  (toByteString . BT.newUserEmailAddress $ newUserBR)
+                  (encodeUtf8 . BT.newUserPassword $ newUserBR)
   pubKey <- fmap expectJust $ liftIO $ readJWK pubKeyPath
   insertedUserIdSCS <- fmap expectRight $ httpSCS $ SCSClient.addUser newUserSCS
-  _insertedUserIdBR <- httpBR $ BRClient.addUser auth newUserBR
   _insertedPrefix <- httpBR $ BRClient.addBusiness auth newBiz
+  _insertedUserIdBR <- httpBR $ BRClient.addUser auth newUserBR
   brKeyId <- fmap expectRight $ httpBR $ BRClient.addPublicKey auth pubKey Nothing
   let basicAuthDataSCS =
         BasicAuthData
           (toByteString . ST.newUserEmailAddress $ newUserSCS)
           (encodeUtf8   . ST.newUserPassword     $ newUserSCS)
 
   let newLocs = flip H.lookup locMap <$> locations
-  traverse_  maybeInsertLocation newLocs
+  traverse_  (maybeInsertLocation userAuth) newLocs
   let updatedHt = H.insert entity (insertedUserIdSCS, basicAuthDataSCS, brKeyId) ht
   insertAndAuth scsUrl brUrl auth locMap updatedHt entities
   where
-    maybeInsertLocation Nothing    = pure ()
-    maybeInsertLocation (Just loc) = void $ runClient brUrl $ BRClient.addLocation auth loc
```

Comment:
Why pattern match instead of `void`?

Reply:
Used void instead :smile: 
Thanks for picking that up